### PR TITLE
fixed base urls for HereMaps

### PR DESCRIPTION
**Pull request by chrstnbwnkl**
passes modified base urls for HereMaps' `.directions()`, `isochrones()` and `matrix()` methods to the instance's client

### CODE REVIEW

File: routingpy/routers/heremaps.py [hunk 1]
```diff
@@ -596,6 +596,9 @@ def directions(  # noqa: C901
             if self.api_key is None
             else "https://route.ls.hereapi.com/routing/7.2"
```
Comment (nilsnolde) [inline routingpy/routers/heremaps.py:597]: this whole block can go actually, it's the same URL as the default from the init

File: routingpy/routers/heremaps.py [hunk 2]
```diff
@@ -596,6 +596,9 @@ def directions(  # noqa: C901
...
         )
+
+        self.client.base_url = self.base_url
```
Comment (nilsnolde) [inline routingpy/routers/heremaps.py:600]: then this is not necessary anymore either

File: routingpy/routers/heremaps.py [hunk 3]
```diff
@@ -997,6 +1000,9 @@ def isochrones(  # noqa: C901
             if self.api_key is None
             else "https://isoline.route.ls.hereapi.com/routing/7.2"
         )
+
+        self.client.base_url = self.base_url
```
Comment (nilsnolde) [inline routingpy/routers/heremaps.py:1004]: skip this and instead assign the URL directly to self.client.base_url

File: routingpy/routers/heremaps.py [hunk 4]
```diff
@@ -1264,6 +1270,8 @@ def matrix(  # noqa: C901
             if self.api_key is None
             else "https://matrix.route.ls.hereapi.com/routing/7.2"
         )
+        self.client.base_url = self.base_url
```
Comment (nilsnolde) [inline routingpy/routers/heremaps.py:1273]: same here

**Review by nilsnolde — COMMENTED** [review]
You're probably not done yet, still my 2c. Also you can actually remove `self.base_url` entirely, no function will use it anymore

File: routingpy/routers/heremaps.py (hunk 1 above)
Comment (chrstnbwnkl) [reply routingpy/routers/heremaps.py:597]: But if you first call `.matrix()` and then `.directions()`, the base_url still needs to change. So set it to a default in the init and then keep the updating block for each of the three methods?

Comment (nilsnolde) [reply routingpy/routers/heremaps.py:597]: yeah you're right! then I guess we can take out the default URLs up top?

File: examples/compare_providers.ipynb [hunk 5]
```diff
@@ -1384,9 +1384,9 @@
  ],
  "metadata": {
   "kernelspec": {
-   "display_name": "routing-py",
```
Comment (nilsnolde) [inline examples/compare_providers.ipynb:1387]: seems unrelated

Comment (chrstnbwnkl) [reply examples/compare_providers.ipynb:1387]: not sure what happened there

File: routingpy/routers/heremaps.py (hunk 1 above)
Comment (chrstnbwnkl) [reply routingpy/routers/heremaps.py:597]: you mean the logic to decide whether to use `_APIKEY_BASE_URL` or `_DEFAULT_BASE_URL`?

Comment (nilsnolde) [reply routingpy/routers/heremaps.py:597]: yep, since we do that now for every single method? unless I'm mistaken there's no need anymore to define self.base_url in the init. and the constants up top defining the URLs are redundant then

Comment (chrstnbwnkl) [reply routingpy/routers/heremaps.py:597]: True, they are. We do have to pass something in the init though, since it's a required arg.

Comment (nilsnolde) [reply routingpy/routers/heremaps.py:597]: if it's not critical for the client's init, simply pass a `''` or just some other garbage;)

File: CHANGELOG.md [hunk 6]
```diff
@@ -12,6 +12,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ### Fixed
 - enhanced error handling for directions with Google [#15](https://github.com/gis-ops/routing-py/issues/15)
 - fixed MapboxOSRM client, removing unused parameters and unifying Isochrones interface [#21](https://github.com/gis-ops/routing-py/issues/21)
+- fixed bug that caused HereMaps client to use wrong API endpoints
```
Comment (nilsnolde) [inline CHANGELOG.md:15]: can you reference the PR at the end of the line?

File: routingpy/routers/heremaps.py [hunk 7]
```diff
@@ -28,8 +28,7 @@
 class HereMaps:
     """Performs requests to the HERE Maps API services."""
 
-    _DEFAULT_BASE_URL = "https://route.api.here.com/routing/7.2"
-    _APIKEY_BASE_URL = "https://route.ls.hereapi.com/routing/7.2"
+    _base_url = ""
```
Comment (nilsnolde) [inline routingpy/routers/heremaps.py:31]: no need for that anymore

File: routingpy/routers/heremaps.py [hunk 8]
```diff
@@ -95,14 +94,12 @@ def __init__(
...
         if self.api_key:
-            self.base_url = self._APIKEY_BASE_URL
             self.auth = {"apikey": self.api_key}
         else:
-            self.base_url = self._DEFAULT_BASE_URL
             self.auth = {"app_id": self.app_id, "app_code": self.app_code}
 
         self.client = client(
-            self.base_url,
+            self._base_url,
```
Comment (nilsnolde) [inline routingpy/routers/heremaps.py:102]: just pass `""` here directly

**Review by nilsnolde — APPROVED** [review]
the last minor stuff and we're good. no need for another review round. you can just fix the few things and merge yourself, I'll approve now.
